### PR TITLE
Add a helper to run only on the selected OS list

### PR DESCRIPTION
**Pull request by ekohl**
This allows us to remove all the duplication in specs.

### CODE REVIEW

**Comment by ekohl** [comment]
@stbenjam this may be interesting for katello-installer-modulesync as well.

File: moduleroot/spec/spec_helper.rb [hunk 1]
```diff
@@ -32,6 +32,14 @@ def exclude_test_os
   end
 end
 
+# Use the above environment variables to limit the platforms under test
+def on_os_under_test
+  on_supported_os.reject do |os, facts|
+    (only_test_os() and not only_test_os.include?(os)) or
```
Comment (domcleal) [inline moduleroot/spec/spec_helper.rb]: Prefer `&&`, `||` and `!` for boolean logic.

Comment (ekohl) [inline moduleroot/spec/spec_helper.rb]: I copied it from our spec files which already include ```and``` and ```not```, but I'll change it.

**Comment by ekohl** [comment]
Updated btw

**Comment by domcleal** [comment]
Thanks @ekohl.

**Pull request merged**